### ISSUE TITLE
Question : scale_x_y lead to more unstable gradients?

### ISSUE THREAD

**Issue by pfeatherstone** [AlexeyAB/darknet#5378]
@AlexeyAB Have you found that using scale_x_y > 1 leads to more unstable gradients early on in training ? I have to admit, I'm not using your repository, but I'm using yolo layers in my model (in Pytorch) and using scale_x_y > 1 combined with CIoU loss leads to unstable gradients. I have to very gradually ramp up the learning rate to avoid this. When scale_x_y==1, I don't have this problem at all.

**Comment by AlexeyAB** [comment]
> Have you found that using scale_x_y > 1 leads to more unstable gradients early on in training ?

How did you check it?

In Pytorch, auto-differentiation and chain-rule is implemented, while in the darknet, I write back propagation manually.
As I remember - I don't calc gradients for `sigmoid*scale_x_y` at all.

**Comment by pfeatherstone** [comment]
> As I remember - I don't calc gradients for sigmoid*scale_x_y at all.

That makes sense.

> How did you check it?

The loss very quickly becomes NaN. Whereas using scale_x_y == 1, loss is never NaN. I'm using AdamW optimizer with a reduceonplateau scheduler. Maybe if I use SGD with a cosine annealing scheduler and long warmup, I won't see this.

Also i'm using scale_x_y == 1.5. Probs a bit high. I noticed you use 1.2, 1.1 and 1.05.

**Comment by AlexeyAB** [comment]
Use 1.1 or 1.2, 1.1 and 1.05.

**Comment by pfeatherstone** [comment]
Used that and had to apply a slow burnin to the learning rate of AdamW optimizer. Works fine and there is a small improvement. Nothing too drastic but enough to warrant the change. Thanks @AlexeyAB
